ArtOfIllusion/src/artofillusion/ArtOfIllusion.java

```diff
@@ -235,6 +219,13 @@ public static ApplicationPreferences getPreferences()
     return preferences;
   }
 
+  /*
+    Method is created for testing purposes to setup preferences at test time.
+  */
+  public static void setPreferences(ApplicationPreferences prefs)
+  {
+    preferences = prefs;
+  }
```

Comment:
This really seems off. Any time that you add a method for "Testing Purposes", figure that you are probably testing the wrong thing.

Reply:
Imagine You want to unit test some new code ex in ViewerCanvas class....
At first You try create instance of this class and only result is NPE because of uninitialized preferences.

Reply:
*If* we need to add such a unit test, and *if* we don't find a better way to pre-configure the test, that would be the appropriate time to add a method. YAGNI.